### ISSUE TITLE
DataTable: should use user's locale for toLowerCase

### ISSUE THREAD

**Issue by georgesimms** [ocadotechnology/hexagonjs#319]
In DataTable (and in fact everywhere where toLowerCase is used), it should be used with the locale in the preferences, which may not be the browser default locale (eg on a shared computer).

**Comment by c-frater** [comment]
Most of the time, we use `toLowerCase` for comparison, rather than for displaying anything to the user so it generally shouldn't be an issue.
The only thing I can think that _could_ be an issue would be where we do comparison on user entered text (e.g. the data table filter) but I can't think of any examples that would cause an issue.
Are there any specific examples that you've found to be an issue?

**Comment by georgesimms** [comment]
Right. If we are only doing it on Computerese, we should make sure it is not using the browser locale (eg to support Turkish I)

**Comment by c-frater** [comment]
Looking into this we should be using `toLocaleLowerCase` for string comparisons in `DataTable`, `Autocomplete` and `Filter` as some locales have special characters with upper and lowercase variants that would currently not display.

This change will most likely be more involved than simply switching over as the browser/user locale may differ from the app locale. 
I'm going to move this out of the Pre-2.0.0 tasks as it is a more involved change that needs more investigation into the best way to implement
